TRUNK-5974 - Use of lucene score for two name queries

fruether opened the pull request:
<!--- Add a pull request title above in this format -->
<!--- real example: 'TRUNK-5111: Replace use of deprecated isVoided' -->
<!--- 'TRUNK-JiraIssueNumber: JiraIssueTitle' -->
## Description of what I changed

This PR is part of a series of commits that wants to enable the Soundex search for Lucene queries within Person search. In this PR I am going to change one of the previous refactorings to use lucene scores.  This dicussion regarding the changes can be found on Talk
https://talk.openmrs.org/t/soundex-search-in-lucenequeries/26068/56
<!--- It can simply be your commit message, which you must have -->

## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first! -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it! -->
<!--- Just add the issue number at the end: -->
see https://issues.openmrs.org/browse/TRUNK-5974

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

Gravewalker666 commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -170,34 +115,29 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	 */
-	private Set<Person> execSoundexLuceneQueryByNames(boolean firstName, String name, Integer birthyear, boolean includeVoided , String gender) {
+	private Set<Person> executeSoundexTwoPersonNamesQuery(String name1, String name2, Integer birthyear, boolean includeVoided , String gender) {
 		PersonLuceneQuery personLuceneQuery = new PersonLuceneQuery(sessionFactory);
```
I think using the firstName and lastName instead of name1 and name2 would give the developers an apparent understanding of what these variables are. But I see you have put the definitions of those params in java comments. Still, I suggest changing them to firstName and lastName for the sake of simplicity. 🙂

fruether replied:
@gravewalker666 Thanks for your feedback. I applied the changes accodingly

fruether replied:
Fixed

HerbertYiga commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -358,14 +297,14 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 		ListPart<Object[]> attributes = attributeQuery.listPartProjection(0, maxResults, "person.personId");
 		attributes.getList().forEach(attribute -> people.add(getPerson((Integer) attribute[0])));
-
+		
```
do we need this?

fruether replied:
Fixed

HerbertYiga commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -330,7 +269,7 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 			return new ArrayList<>();
 		}
-
+		
```
do we need this?

fruether replied:
Done

HerbertYiga commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -330,7 +269,7 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 			return new ArrayList<>();
 		}
-
+		
```
do we need this?

fruether replied:
Fixed

HerbertYiga commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -344,11 +283,11 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 				criteria.add(Restrictions.eq("personVoided", false));
 			}
-
+			
```
do we need this?

fruether replied:
Done

HerbertYiga commented on api/src/main/java/org/openmrs/api/db/hibernate/PersonLuceneQuery.java:
```diff
@@ -273,14 +225,14 @@ public PersonLuceneQuery(SessionFactory sessionFactory) {
 			luceneQuery.useOrQueryParser();
 		}
-
+		
```
do we need this?

fruether replied:
Fixed

HerbertYiga commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -589,9 +528,9 @@ public Relationship getRelationship(Integer relationshipId) throws DAOException
+		
 		return (RelationshipType) sessionFactory.getCurrentSession().get(
-		    RelationshipType.class, relationshipTypeId);
+			RelationshipType.class, relationshipTypeId);
```
did you make any change here?

HerbertYiga commented:
@Gravewalker666 can you remove all the unwanted changes from your code

Gravewalker666 commented:
@HerbertYiga I think you meant to tag @fruether 
No worries tho 😄

fruether commented:
@HerbertYiga I worked on it and removed a lot space and tabs. Some are still missing I will work on that. Feeld free to review in the maintime the actual code I changed!

fruether commented:
I don't know why. It runs locally on my machine and I did not change anything related to the failing tests. @HerbertYiga @Gravewalker666 any chance you can try to run it locally

dkayiwa commented on api/src/main/java/org/openmrs/api/db/hibernate/PersonLuceneQuery.java:
```diff
@@ -124,54 +96,35 @@ public PersonLuceneQuery(SessionFactory sessionFactory) {
+	 * @param birthyear the birthyear the searched person should have 
 	 * @param includeVoided is true if voided person should be matched
-	 * @param gender the gender of the person to search
-	 * @return the LuceneQuery that returns Persons with a soundex representation of the secondName and other defined search criteria
```
What is the reason for replacing `secondName` with `firstName`?

fruether commented:
@dkayiwa I adapted the comment!

dkayiwa commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -170,34 +115,28 @@ public void setSessionFactory(SessionFactory sessionFactory) {
+	 * @return the set of Persons that match the search criteria
 	 */
-	private Set<Person> execSoundexLuceneQueryByNames(boolean firstName, String name, Integer birthyear, boolean includeVoided , String gender) {
+	private Set<Person> executeSoundexTwoPersonNamesQuery(String firstName, String lastName, Integer birthyear, boolean includeVoided , String gender) {
```
What is the reason for changing from `String name1, String name2` to `String firstName, String lastName`?

fruether commented:
Because the previous reviewer asked me to to so as that would result to
more readable code. I agreed to that feedback and changed it

dkayiwa ***@***.***> schrieb am Di., 27. Apr. 2021, 15:18:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java
> <https://github.com/openmrs/openmrs-core/pull/3738#discussion_r621198930>:
>
> >  	 */
> -	private Set<Person> execSoundexLuceneQueryByNames(boolean firstName, String name, Integer birthyear, boolean includeVoided , String gender) {
> +	private Set<Person> executeSoundexTwoPersonNamesQuery(String firstName, String lastName, Integer birthyear, boolean includeVoided , String gender) {
>
> What is the reason for changing from String name1, String name2 to String
> firstName, String lastName?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/3738#pullrequestreview-645842924>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMEOAACKHNAVSWAMYFJ6DLTK22Q3ANCNFSM42KUKSCA>
> .
>

dkayiwa commented:
Was the goal to match with OpenMRS's familyName,middleName and givenName?
Where givenName = firstName and familyName = lastName

fruether commented:
Yes only that the matching is not perfect as n2 could represent a family
name or middle name. That's probably why in the original code it was
referred to just as n1 and n2 but which is harder to reason about.

Not sure if middle name classifieds as last name

dkayiwa ***@***.***> schrieb am Di., 27. Apr. 2021, 15:41:

> Was the goal to match with OpenMRS's familyName,middleName and givenName?
> Where givenName = firstName and familyName = lastName
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/3738#issuecomment-827615586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMEOAFV6HR5HJHEUU7DHHLTK25JBANCNFSM42KUKSCA>
> .
>

dkayiwa commented:
In that case don't you find those names more confusing than the original name1, name2, etc?

fruether commented:
I applied the Reviewer comment from before. In my opinion: searchName1,
searchName2, searchName3 would be best suited

Am Di., 27. Apr. 2021 um 15:52 Uhr schrieb dkayiwa ***@***.***
>:

> In that case don't you find those names more confusing than the original
> name1, name2, etc?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/3738#issuecomment-827623966>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMEOAGHL5NBM67BSCACZDTTK26TFANCNFSM42KUKSCA>
> .
>

dkayiwa commented:
Ok let us change it to your proposal.

fruether commented:
@dkayiwa I adapted the naming accordingly. So the discussed issue is resolved

dkayiwa commented on api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:
```diff
@@ -170,34 +115,28 @@ public void setSessionFactory(SessionFactory sessionFactory) {
+	 * @return the set of Persons that match the search criteria
 	 */
-	private Set<Person> execSoundexLuceneQueryByNames(boolean firstName, String name, Integer birthyear, boolean includeVoided , String gender) {
+	private Set<Person> executeSoundexTwoPersonNamesQuery(String searchName1, String searchName2, Integer birthyear, boolean includeVoided , String gender) {
```
Could you remind me as to why we are replacing the firstName boolean parameter with a String searchName1?

fruether commented:
@dkayiwa it was previously a String as well. However, Git mapped it to another line where it was a booles. The method names are different too as they are different methods
